src/app/core/layout/header/header.component.ts

```diff
@@ -23,24 +24,17 @@ import { MainNavigationComponent } from '../../components/main-navigation/main-n
 })
 export class HeaderComponent {
   isLoggedIn = false;
-  displayName = 'Account';
+  displayName$?: Observable<string | undefined>;
   menuItems!: MenuItem[];
 
   constructor(
     private authenticationService: AuthenticationService,
     private store: Store,
   ) {
-    this.authenticationService
-      .getUser()
-      .pipe(takeUntilDestroyed())
-      .subscribe((user) => {
-        if (user) {
-          this.isLoggedIn = true;
-          this.displayName = user.displayName ?? (user.email as string);
-        } else {
-          this.isLoggedIn = false;
-          this.displayName = 'Account';
-        }
+    this.displayName$ = this.authenticationService.getUser().pipe(
+      takeUntilDestroyed(),
+      tap((user) => {
```

Comment:
Doesn't feel good.
Your `this.displayName$` transformation is ideal. I was cheering at it because you use an actual stream of data with your template instead of setting a variable. Perfect 🙌

But why sneaking in that tap, that side effect? Make it another stream `menuItems` or such. 
rxjs + modifying local varaibles is a bad pattern.